mono/eglib/glib.h

```diff
@@ -107,6 +107,56 @@ struct g_cast
 
 #endif
 
+#ifdef __cplusplus
+
+// Provide for bit operations on enums, but not all integer operations.
+// This alleviates a fair number of casts in porting C to C++.
+
+// Forward declare template with no generic implementation.
+template <size_t> struct mono_size_to_int;
+
+// Template specializations.
+template <> struct mono_size_to_int<1> { typedef int8_t T; };
+template <> struct mono_size_to_int<2> { typedef int16_t T; };
+template <> struct mono_size_to_int<4> { typedef int32_t T; };
+template <> struct mono_size_to_int<8> { typedef int64_t T; };
+
+#define MONO_SIZE_TO_INT(x) mono_size_to_int <sizeof (x)>::T
+
+#define G_ENUM_BINOP(Enum, op, opeq) 		\
+inline Enum					\
+operator op (Enum a, Enum b)			\
+{						\
+	typedef MONO_SIZE_TO_INT (a) T;		\
+	return (Enum)((T)a op (T)b);		\
```

Comment:
These casts could be `static_cast<>`.

Reply:
I was erring toward less C++-y,  but yes, will fix.